### ISSUE TITLE
examples/edge: ImageParts draws outside pixels unexpectedly even with nearest filter

### ISSUE THREAD

**Issue by hajimehoshi** [hajimehoshi/ebiten#317]
https://gyazo.com/009fcc1e4666f4f0ef407e8b610bbd92

The green part is not included by the indicated image parts, but was drawn when the angle is 45° or 135° (at least on macOS). I think this is expected when the filter is linear but this should not happen with nearest filter.

**Comment by hajimehoshi** [comment]
Reference to understand coordinate system on OpenGL http://hacksoflife.blogspot.jp/2009/12/texture-coordinate-system-for-opengl.html

**Comment by hajimehoshi** [comment]
Adjusting texel coordinates fixes this issue, but I'm not sure '16384' is an appropriate value... (32768 didn't fix this).

```diff
diff --git a/vertices_js.go b/vertices_js.go
index 759033a..9db9cc8 100644
--- a/vertices_js.go
+++ b/vertices_js.go
@@ -56,6 +56,8 @@ func vertices(parts ImageParts, width, height int, geo *affine.GeoM) []float32 {
                        continue
                }
                u0, v0, u1, v1 := float64(sx0)/wf, float64(sy0)/hf, float64(sx1)/wf, float64(sy1)/hf
+               u1 -= 1.0/16384.0
+               v1 -= 1.0/16384.0
                vs.SetIndex(n, dx0)
                vs.SetIndex(n+1, dy0)
                vs.SetIndex(n+2, u0)
diff --git a/vertices_notjs.go b/vertices_notjs.go
index febc9bf..b3e57e6 100644
--- a/vertices_notjs.go
+++ b/vertices_notjs.go
@@ -55,6 +55,8 @@ func vertices(parts ImageParts, width, height int, geo *affine.GeoM) []float32 {
                        continue
                }
                u0, v0, u1, v1 := float32(sx0)/wf, float32(sy0)/hf, float32(sx1)/wf, float32(sy1)/hf
+               u1 -= 1.0/16384.0
+               v1 -= 1.0/16384.0
                vs[n] = x0
                vs[n+1] = y0
                vs[n+2] = u0
```

**Comment by hajimehoshi** [comment]
I have checked this on MacBook Pro and Inspiron 11 3162. I'll check this on Vaio Z Canvas later.

**Comment by hajimehoshi** [comment]
Confirmed this on Vaio Z Canvas. On all environments, I have confirmed that this bug can be reproduced before the fix and bb993a4 fixed that. Let me close this.
